messagingpush/src/main/java/io/customer/messagingpush/network/HTTPClient.kt

```diff
@@ -0,0 +1,120 @@
+package io.customer.messagingpush.network
+
+import android.util.Base64
+import io.customer.sdk.core.di.SDKComponent
+import io.customer.sdk.core.util.DispatchersProvider
+import io.customer.sdk.data.store.Client
+import io.customer.sdk.data.store.GlobalPreferenceStore
+import java.io.IOException
+import java.net.HttpURLConnection
+import java.net.MalformedURLException
+import java.net.URL
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+internal data class HttpRequestParams(
+    val path: String,
+    val headers: Map<String, String> = emptyMap(),
+    val body: String? = null
+)
+
+internal interface HttpClient {
+    /**
+     * Performs a POST request to [params.url] with [params.headers] and [params.body].
+     *
+     * @param params The request parameters (URL, headers, body).
+     * @param onComplete Callback invoked with a `Result<String>`:
+     *   - `Result.success(responseBody)` for 2xx response codes
+     *   - `Result.failure(exception)` for network errors or non-2xx codes
+     */
+    fun request(params: HttpRequestParams, onComplete: (Result<String>) -> Unit)
+}
+
+internal class HttpClientImpl : HttpClient {
+
+    private val connectTimeoutMs = 15_000
+    private val readTimeoutMs = 20_000
```

Comment:
Do we really need such high timeout values here? 🤔

Reply:
nope not really, got these from iOS HTTP client and then kept it for consistency. We can lower it if needed.